daemon/codec.c

```diff
@@ -3992,8 +3993,16 @@ static void packet_encoded_tx(AVPacket *pkt, struct codec_ssrc_handler *ch, stru
 		payload_type = dtmf_pt;
 		if (is_dtmf == 1)
 			ch->rtp_mark = 1; // DTMF start event
-		else if (is_dtmf == 3)
-			repeats = 2; // DTMF end event
+		else {
+			// we need to pass a ts_delay to codec_output_rtp to ensure the calculated time
+			// to send the packet is offset by the event duration of the DTMF packets
+			// but we need to reduce it by one packet duration so that the delay is offset
+			// from the first event packet
+			struct telephone_event_payload *ev_pt = (void *) inout->s;
+			ts_delay = ntohs(ev_pt->duration) - (ch->handler->dest_pt.ptime * ch->handler->dest_pt.clock_rate / 1000);
+			if (is_dtmf == 3)
+				repeats = 2; // DTMF end event
+		}
```

Comment:
Do you think it would make sense to always do this TS adjustment, and not just for non-start event packets? I assume in your case the delay probably comes out as zero for the initial packet, but it's possible the first packet (triggering the start event) uses a different duration.

Reply:
I don't think so. the first event packet has had its ts increased from the previous packet already so it gets scheduled correctly; in the pcaps I made during testing, the first packet always looked right (~20ms from the previous non-event packet). Then, because the `ts_delay` is applied on top of that first event packet TS, this adjustment has to be one ptime less than the current event duration so that they don't end up being scheduled starting 40ms after the first event.

Reply:
> I don't think so. the first event packet has had its ts increased from the previous packet already so it gets scheduled correctly; in the pcaps I made during testing, the first packet always looked right (~20ms from the previous non-event packet). Then, because the `ts_delay` is applied on top of that first event packet TS, this adjustment has to be one ptime less than the current event duration so that they don't end up being scheduled starting 40ms after the first event.

AFAICT the first event packet _usually_ has a duration of 20 ms or 160 samples, so that would come out as a delay of 160 - 20 * 8 = 0. But what if that packet got lost, and the packet triggering the start event (i.e. the state change) is actually the second one, using a delay of 40 ms or 320 samples?

Reply:
i gone ahead and done that anyway :)